regression/validate-trace-xml-schema/check.py

```diff
@@ -0,0 +1,172 @@
+
+TraceXsdSpec = os.path.abspath(os.path.join(this_script_dir, '..', '..', 'doc/assets/xml_spec.xsd'))
+
+test_base_dir = os.path.abspath(os.path.join(this_script_dir, '..', 'cbmc'))
+
+# some tests in the cbmc suite don't work for the trace checks for one reason or another
+ExcludedTests = list(map(lambda s: os.path.join(test_base_dir, s), [
+    # these tests except input from stdin
+    'json-interface1/test_wrong_option.desc',
+    'json-interface1/test.desc',
+    'json-interface1/test_wrong_flag.desc',
+    'xml-interface1/test_wrong_option.desc',
+    'xml-interface1/test.desc',
+    'xml-interface1/test_wrong_flag.desc',
+    # these want --show-goto-functions instead of producing a trace
+    'destructors/compound_literal.desc',
+    'destructors/enter_lexical_block.desc',
+    'reachability-slice-interproc2/test.desc',
+    # this one wants show-properties instead producing a trace
+    'show_properties1/test.desc',
+    # program-only instead of trace
+    'vla1/program-only.desc',
+    # this one doesn't work for me locally at all
+    'integer-assignments1/test.desc',
+    # these test for invalid command line handling
+    'bad_option/test_multiple.desc',
+    'bad_option/test.desc',
+    # this one produces XML intermingled with main XML output when used with --xml-ui
+    'graphml_witness2/test.desc',
+    # this one uses program-only again
+    'Quantifiers-simplify/simplify_not_forall.desc',
+    # produces intermingled XML on the command line
+    'coverage_report1/test.desc',
+    'coverage_report1/paths.desc',
+    # this uses json-ui (fails for a different reason actually, but should also
+    #   fail because of command line incompatibility)
+    'json1/test.desc',
+    # uses show-goto-functions
+    'reachability-slice/test.desc',
+    'reachability-slice/test2.desc',
+    'reachability-slice/test3.desc',
+    'reachability-slice-interproc/test.desc',
+    # again, intermingles several XML documents
+    'graphml_witness1/test.desc',
+    'switch8/program-only.desc',
+    # show-goto-functions
+    'unsigned1/test.desc',
+    # show-goto-functions
+    'reachability-slice-interproc3/test.desc',
+    'sync_lock_release-1/symbol_per_type.desc'
+]))
+
+if len(sys.argv) != 2:
+    sys.stderr.write('Usage: check.py <path-to-cbmc>')
+CbmcPath = os.path.abspath(sys.argv[1])
+XsdValidateJar = os.path.join(this_script_dir, 'validate-xsd/build/libs/validate-xsd-1.0-SNAPSHOT-uber.jar')
+
+
+class ChangeDir:
+    def __init__(self, change_to_wd):
+        self.current_wd = os.getcwd()
+        self.change_to_wd = change_to_wd
+
+    def __enter__(self):
+        os.chdir(self.change_to_wd)
+
+    def __exit__(self, _type, _value, _tb):
+        os.chdir(self.current_wd)
+
+
+class TestSpec:
+    def __init__(self, args, is_knownbug, is_future, is_thorough):
+        self.args = args
+        self.is_knownbug = is_knownbug
+        self.is_future = is_future
+        self.is_thorough = is_thorough
```

Comment:
I think we've written a `.desc` python parser before (possibly for test-gen) :bulb: Pull out that into a separate python module so that it may be reused

Reply:
For reference, we [did](https://github.com/diffblue/cover/blob/develop/test-gen/regression/common.py#L50 :lock:),  but not trivial to unify these two at the moment.